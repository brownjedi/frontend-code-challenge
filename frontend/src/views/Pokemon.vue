<template>
  <div :class="styles.container">
    <Details :name="name" />
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, useCssModule } from 'vue'
import { useRoute } from 'vue-router'
import Details from '@/components/Details.vue'

export default defineComponent({
  setup() {
    const styles = useCssModule()
    const route = useRoute()
    const name = ref(route.params.name as string)

    return {
      name,
      styles,
    }
  },
  components: {
    Details,
  },
})
</script>

<style lang="scss" module>
.container {
}
</style>
