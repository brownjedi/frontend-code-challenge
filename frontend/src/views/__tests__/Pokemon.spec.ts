import { shallowMount } from '@vue/test-utils'
import Pokemon from '@/views/Pokemon.vue'

const routeMock = jest.fn()
jest.mock('vue-router', () => ({
  useRoute: () => routeMock(),
}))

describe('views/Pokemon.vue', () => {
  beforeEach(() => jest.resetAllMocks())

  it('has the expected html structure', () => {
    routeMock.mockImplementationOnce(() => ({
      params: {
        name: 'Bulbasaur',
      },
    }))

    const wrapper = shallowMount(Pokemon)
    expect(wrapper.element).toMatchSnapshot()
  })
})
