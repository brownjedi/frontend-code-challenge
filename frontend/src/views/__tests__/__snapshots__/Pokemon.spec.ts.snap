// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/Pokemon.vue has the expected html structure 1`] = `
<div
  class=""
>
  <details-stub
    name="Bulbasaur"
  />
</div>
`;
